iroh-p2p/src/behaviour.rs

```diff
@@ -34,6 +36,10 @@ pub(crate) struct NodeBehaviour {
     bitswap: Bitswap,
     pub(crate) kad: Toggle<Kademlia<MemoryStore>>,
     mdns: Toggle<Mdns>,
+    pub(crate) autonat: Toggle<autonat::Behaviour>,
+    relay: Toggle<relay::v2::relay::Relay>,
+    relay_client: Toggle<relay::v2::client::Client>,
+    dcutr: Toggle<dcutr::behaviour::Behaviour>,
```

Comment:
Ideally we would have an `AndBehaviour`, that way you could do:

``` rust
hole_punching_client: Toggle<AndBehaviour<relay::v2::client::Client, dcutr::behaviour::Behaviour>>`
```

That way it is enforced at compile time that you always either enable both or disable both.

Reply:
yep, that would would be nice

Reply:
:+1: Tracked here: https://github.com/libp2p/rust-libp2p/issues/2719